src/main/java/ch/njol/skript/expressions/ExprLastDeathTime.java

```diff
@@ -0,0 +1,181 @@
+		setExpr((Expression<Player>) exprs[0]);
+		return true;
+	}
+
+	@Nullable
+	@Override
+	public Date convert(Player player) {
+		long ticksSinceDeath = player.getStatistic(Statistic.TIME_SINCE_DEATH);
+		if (ticksSinceDeath < 0) 
+			return null;
+		Timespan timeSinceDeath = Timespan.fromTicks_i(ticksSinceDeath);
+		return Date.now().minus(timeSinceDeath);
+	}
+
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		// Use a date for setting (e.g. 'set last death time of player to 1 hour ago')
+		if (mode == ChangeMode.SET)
+			return CollectionUtils.array(Date.class);
+		// Use a timespan for adding/remove (e.g 'add 10 seconds to the time of the player's last death')
+		if (mode == ChangeMode.ADD || mode == ChangeMode.REMOVE)
+			return CollectionUtils.array(Timespan.class);
+		return null;
+	}
+
+	/*
+	 * Date#compare(date)
+	 * This method returns whether the passed date is
+	 * before, the same as, or after the date it's being compared to.
+	 * A value less than 0 indicates that the passed date is AFTER the date it's being compared to.
+	 * A value of 0 indicates that the passed date is the SAME the date it's being compared to.
+	 * A value greater than 0 indicates that passed date is BEFORE the date it's being compared to.
+	 * 
+	 * We use this method because we don't want the new death date to be in the future
+	 */
+	@SuppressWarnings("null")
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+
+		if (delta == null)
+			return;
+
+		Date now = Date.now();
+
+		for (Player player : getExpr().getArray(e)) {
+			if (mode == ChangeMode.ADD) {
+
+				// Get the current death time of the player.
+				Date deathTime = convert(player);
+
+				deathTime.add((Timespan) delta[0]);
+
+				if (deathTime.compareTo(now) < 1) {
+					// Get the timespan representing the new time since death (in ticks).
+					long newTimespanInTicks= deathTime.difference(now).getTicks_i();
+
+					/*
+					 *  Convert to int.
+					 *  If it is greater than the max value for an integer,
+					 *  the user probably wants to set it to the max value, right?
+					 */
+					int newTimeSinceDeath = newTimespanInTicks < Integer.MAX_VALUE ? (int) newTimespanInTicks : Integer.MAX_VALUE;
+
+					player.setStatistic(Statistic.TIME_SINCE_DEATH, newTimeSinceDeath);
+				}
+
+			} else if (mode == ChangeMode.REMOVE) {
+
+				// Get the current death time of the player.
+				Date deathTime = convert(player);
+
+				deathTime.subtract((Timespan) delta[0]);
+				
+				// Get the timespan representing the new time since death (in ticks).
+				long newTimespanInTicks= deathTime.difference(now).getTicks_i();
+
+				/*
+				 *  Convert to int.
+				 *  If it is greater than the max value for an integer,
+				 *  the user probably wants to set it to the max value, right?
+				 */
+				int newTimeSinceDeath = newTimespanInTicks < Integer.MAX_VALUE ? (int) newTimespanInTicks : Integer.MAX_VALUE;
```

Comment:
you could also do `int newTimeSinceDeath = Math.min(newTimespanInTicks, Integer.MAX_VALUE);`